.github/actions/wait-docker-ready/action.yaml

```diff
@@ -8,11 +8,14 @@ runs:
     - name: Wait for Docker daemon to be ready
       shell: bash
       run: |-
-          bash -c '
-          for i in {1..150}
-          do
-            docker info &> /dev/null && exit
-            sleep 2
-          done
-          echo "Could not reach Docker daemon from buildbot worker" >&2
-          exit 1'
+        bash -c '
+        for i in {1..150}
+        do
+          docker info &> /dev/null && exit
+          if [ $i -eq 50 ]; then
+            sudo systemctl restart docker
```

Comment:
that seems really fishy : docker daemon in CI is stable as it gets, there is no reason we would need to restart it....
(even waiting for it is weird : we have many CI jobs which build images, and I don't think we need this kind of retry/restart of docker)

→ seems like this whole step is a workaround for another issue, would be better to investigate/fix that issue instead of burying it

Reply:
You may be right indeed, but looking at this, for example: https://github.com/scality/Zenko/actions/runs/4741155549/jobs/8417897238#step:4:163 does not show anything bad beyond the fact that we timeout trying to contact the docker daemon (usually it takes "0s" to complete). Hence this try to restart the daemon, in case something was preventing it from starting...